File: bwl.Imaging/MatrixProcessing/Extensions.vb
Imports System.Drawing
Imports System.Runtime.CompilerServices

Public Module Extensions
    <Extension()>
    Public Function ToPoint(pointF As PointF) As System.Drawing.Point
        Return New System.Drawing.Point(CInt(pointF.X), CInt(pointF.Y))
    End Function

    <Extension()>
    Public Function ToPointF(point As Point) As System.Drawing.PointF
        Return New System.Drawing.PointF(point.X, point.Y)
    End Function

    <Extension()>
    Public Function Dist(this As PointF, arg As PointF) As Single
        Return CSng(Math.Sqrt((this.X - arg.X) ^ 2 + (this.Y - arg.Y) ^ 2))
    End Function

    <Extension()>
    Public Function ToString(this As PointF) As String
        Return "X: " + this.X.ToString() + " Y: " + this.Y.ToString
    End Function

    <Extension()>
    Public Function ToPositiveSized(this As RectangleF) As RectangleF
        Dim left = Math.Min(this.Left, this.Right)
        Dim right = Math.Max(this.Left, this.Right)
        Dim top = Math.Min(this.Top, this.Bottom)
        Dim bottom = Math.Max(this.Top, this.Bottom)
        Return RectangleF.FromLTRB(left, top, right, bottom)
    End Function

    <Extension()>
    Public Function ToPositiveSized(this As Rectangle) As Rectangle
        Dim left = Math.Min(this.Left, this.Right)
        Dim right = Math.Max(this.Left, this.Right)
        Dim top = Math.Min(this.Top, this.Bottom)
        Dim bottom = Math.Max(this.Top, this.Bottom)
        Return Rectangle.FromLTRB(left, top, right, bottom)
    End Function
End Module

File: Bwl.Imaging/MatrixProcessing/Extensions.cs
using System;
using System.Drawing;

namespace Bwl.Imaging
{

    public static class Extensions
    {
        public static Point ToPoint(this PointF pointF)
        {
            return new Point((int)Math.Round(pointF.X), (int)Math.Round(pointF.Y));
        }

        public static PointF ToPointF(this Point point)
        {
            return new PointF(point.X, point.Y);
        }

        public static float Dist(this PointF @this, PointF arg)
        {
            return (float)Math.Sqrt(Math.Pow((double)(@this.X - arg.X), 2d) + Math.Pow((double)(@this.Y - arg.Y), 2d));
        }

        public static string ToString(this PointF @this)
        {
            return "X: " + @this.X.ToString() + " Y: " + @this.Y.ToString();
        }

        public static RectangleF ToPositiveSized(this RectangleF @this)
        {
            float left = Math.Min(@this.Left, @this.Right);
            float right = Math.Max(@this.Left, @this.Right);
            float top = Math.Min(@this.Top, @this.Bottom);
            float bottom = Math.Max(@this.Top, @this.Bottom);
            return RectangleF.FromLTRB(left, top, right, bottom);
        }

        public static Rectangle ToPositiveSized(this Rectangle @this)
        {
            int left = Math.Min(@this.Left, @this.Right);
            int right = Math.Max(@this.Left, @this.Right);
            int top = Math.Min(@this.Top, @this.Bottom);
            int bottom = Math.Max(@this.Top, @this.Bottom);
            return Rectangle.FromLTRB(left, top, right, bottom);
        }
    }
}